internal/hooks/module_source_registry.go

```diff
@@ -0,0 +1,86 @@
+package hooks
+
+import (
+	"context"
+	"strings"
+
+	"github.com/algolia/algoliasearch-client-go/v3/algolia/opt"
+	"github.com/hashicorp/hcl-lang/decoder"
+	"github.com/hashicorp/hcl-lang/lang"
+	"github.com/zclconf/go-cty/cty"
+)
+
+type RegistryModule struct {
+	FullName    string `json:"full-name"`
+	Description string `json:"description"`
+}
+
+func (h *Hooks) fetchModulesFromAlgolia(ctx context.Context, term string) ([]RegistryModule, error) {
+	modules := make([]RegistryModule, 0)
+
+	index := h.AlgoliaClient.InitIndex("tf-registry:prod:modules")
+	params := []interface{}{
+		ctx, // transport.Request will magically extract the context from here
+		opt.AttributesToRetrieve("full-name", "description"),
+		opt.HitsPerPage(10),
+	}
+
+	res, err := index.Search(term, params...)
+	if err != nil {
+		return modules, err
+	}
+
+	err = res.UnmarshalHits(&modules)
+	if err != nil {
+		return modules, err
+
+	}
+
+	return modules, nil
+}
+
+func (h *Hooks) RegistryModuleSources(ctx context.Context, value cty.Value) ([]decoder.Candidate, error) {
+	candidates := make([]decoder.Candidate, 0)
+	prefix := value.AsString()
+
+	if isModuleSourceLocal(prefix) {
+		// We're dealing with a local module source here, no need to search the registry
+		return candidates, nil
+	}
+
+	if h.AlgoliaClient == nil {
+		return candidates, nil
+	}
+
+	modules, err := h.fetchModulesFromAlgolia(ctx, prefix)
+	if err != nil {
+		return candidates, err
```

Comment:
Given that we ignore these errors from hooks upstream https://github.com/hashicorp/hcl-lang/blob/b66451713db4e6e3bcf6cde7d70bd8cb51cc2729/decoder/expression_candidates.go#L278

is it worth logging it here? e.g. by plumbing logger to `Hooks` and using it here.